conda/gateways/disk/read.py

```diff
@@ -45,17 +46,44 @@ def collect_all_info_for_package(extracted_package_directory):
     info_dir = join(extracted_package_directory, 'info')
 
     files_path = join(extracted_package_directory, 'info', 'files')
-    files = tuple(ln for ln in (line.strip() for line in yield_lines(files_path)) if ln)
+    file_json_path = join(extracted_package_directory, 'info', 'files.json')
+
+    if isfile(file_json_path):
+        with open(file_json_path) as file_json:
+            data = json.load(file_json)
+        if data.get('version') != 1:
+            raise CondaUpgradeError("Expected files.json schema to be version 1")
+        return PackageInfo(files=(PathInfoV1(**f) for f in data['files']),
```

Comment:
Just a small thing here... With all of these named arguments, I've found it better to create them one-per-line outside of the instance object creation.  That way, if/when you get a stack trace, you can more easily tell which of all of these function it's actually coming from.  So what I mean is something like

```
files=(PathInfoV1(**f) for f in data['files'])
index_json_record=read_index_json(extracted_package_directory)
noarch=read_noarch(extracted_package_directory)
icondata=read_icondata(extracted_package_directory)

return PackageInfo(files=files, index_json_record=index_json_record ...)
```